providers/dns/constellix/constellix.go

```diff
@@ -122,7 +122,10 @@ func (d *DNSProvider) Present(domain, token, keyAuth string) error {
 
```

Comment:
These ~20 lines at the start are almost identical to the first ~20 lines of `CleanUp()`. Maybe we can DRY'up and move it into a helper method? Something like

```go
func (d *DNSProvider) preflightcheck(domain, keyAuth string) (authZone string, dom internal.Domain, records []internal.Records, err error) {
	fqdn, value := dns01.GetRecord(domain, keyAuth)

	authZone, err = dns01.FindZoneByFqdn(fqdn)
	if err != nil {
		err = fmt.Errorf("constellix: could not find zone for domain %q and fqdn %q : %w", domain, fqdn, err)
		return
	}

	dom, err = d.client.Domains.GetByName(dns01.UnFqdn(authZone))
	if err != nil {
		err = fmt.Errorf("constellix: failed to get domain: %w", err)
		return
	}

	recordName := getRecordName(fqdn, authZone)

	records, err = d.client.TxtRecords.Search(dom.ID, internal.Exact, recordName)
	if err != nil {
		err = fmt.Errorf("constellix: failed to search TXT records: %w", err)
		return
	}

	if len(records) > 1 {
		err = errors.New("constellix: failed to get TXT records")
		return err
	}
	return
}

func (d *DNSProvider) Present(domain, token, keyAuth string) error {
	authZone, dom, records, err := preflightcheck(domain, keyAuth string)
	if err != nil {
		return err
	}
	// ...
}
```

I'm usually not a fan of methods returning more than three values, but adding a `type preflight struct { authZone string, dom internal.Domain, records []internal.Records }` wouldn't clear things up by that much, wouldn't it?

Reply:
I'm usually not a fan of technical factorization: I prefer "business/behavioral" factorization.
I'm also not a use fan of returning more than three values, named returns.

`preflight`method and struct are just a technical factorization, it can a be seen as a simplification from a maintenance point of view but also as a limitation of the capacities of evolution.

So I would prefer to extract methods from the "second" parts of the functions, because those sections are more "business" related (create/delete TXT records)

Reply:
Sounds good to me :+1: